source/userguide/auto_annotator.rst

```diff
@@ -0,0 +1,456 @@
+	is it correct to say that {Angela|name} is a {boss|manager}?
+	who all is {management|manager} of {tayana|name}
+
+After running :attr:`annotate` we find that instances of :attr:`sys_person` have been labelled and have overwritten previous instances of the custom entity, :attr:`name`.
+
+.. code-block:: none
+
+	I wanna get a list of all of the employees that are currently {manage|manager} {caroline|sys_person}
+	I wanna know {Tayana Jeannite|sys_person}'s person in {leadership|manager} of her?
+	is it correct to say that {Angela|sys_person} is a {boss|manager}?
+	who all is {management|manager} of {tayana|sys_person}
+
+
+Creating a Custom Annotator
+---------------------------
+The :class:`SpacyAnnotator` is a subclass of the abstract base class :class:`Annotator`.
+The functionality for annotating and unannotating files is contained in :class:`Annotator` itself.
+A developer simply needs to implement two methods to create a custom annotator.
+
+
+Custom Annotator Boilerplate Code
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+This section includes boilerplate code to build a :class:`CustomAnnotator` class to add to :attr:`auto_annotator.py`
+There are two "TODO"s. To implement a :class:`CustomAnnotator` class a developer has to implement the :meth:`parse` and :meth:`valid_entity_check` methods.
+
+.. code-block:: python
+
+	class CustomAnnotator(Annotator):
+		""" Custom Annotator class used to generate annotations.
+		"""
+
+		def __init__(self, app_path, config=None):
+			super().__init__(app_path=app_path, config=config)
+			
+			# Add additional attributes if needed
+
+		def parse(self, sentence, entity_types=None):
+			""" 
+			Args:
+				sentence (str): Sentence to detect entities.
+				entity_types (list): List of entity types to parse. If None, all
+					possible entity types will be parsed.
+			Returns: entities (list): List of entity dictionaries.
+			"""
+
+			# TODO: Add custom parse logic
+
+			return entities
+
+		def valid_entity_check(self, entity):
+			""" 
+			Args: entity (str): Name of entity to annotate.
+			Returns: bool: Whether entity is valid.
+			"""
+			entity = entity.lower().strip()
+			supported_entities = [
+
+			# TODO: Add list of supported entity names
+			
+			]
+			return entity in supported_entities
+
+Entities returned by :attr:`parse()` must have the following format:
+
+.. code-block:: python
+
+	entity = { 
+		"body": (substring of sentence), 
+		"start": (start index), 
+		"end": (end index + 1), 
+		"dim": (entity type), 
+		"value": (resolved value, if it exists), 
+	}
+
+Registering Annotator to Enable command-line Use
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In order to use our :class:`CustomAnnotator` when running annotation/unannotation commands from the command-line, we must add the following line to the bottom of the :attr:`auto_annotator.py` file to register the class and ensure that it loads in the :attr:`cli.py` file.
```

Comment:
Ideally for a person to implement their own custom annotator, they should not add to MindMeld's source code. Thoughts?

Reply:
That's a good point. We could make the cli command exclusive to the mindmeld inbuilt SpacyAnnotator and potential future mindmeld Annotators.

Developers can run their annotators in the same file with a small addition to the boiler plate code presented in the docs.

if __name__ == "__main__":
    custom_annotator = CustomAnnotator(app_path="hr_assistant")
    custom_annotator.annotate()

Thoughts on this approach?

Reply:
Yup I think that should work